src/detect-bsize.c

```diff
@@ -291,6 +330,13 @@ static int DetectBsizeSetup (DetectEngineCtx *de_ctx, Signature *s, const char *
     DetectBsizeData *bsz = DetectBsizeParse(sizestr);
     if (bsz == NULL)
         goto error;
+
+    if (s->init_data->smlists[list] != NULL) {
```

Comment:
by putting this here, the validation happens while new conditions can still be added. What happens with:
`http.uri; content:"abc"; bsize:3; content:"abcdef";`?